### ISSUE TITLE
Add tests against Python 3.9 and claim support for it in our packages

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#8134]
We should at least do this once it is released which the [release schedule](https://www.python.org/dev/peps/pep-0596/#release-schedule) expects to be October 5th.

As of writing this, Azure Pipelines does not offer a prerelease version of Python 3.9. You can check the included software on each OS in Azure at https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software.

**Comment by FelixSchwarz** [comment]
Just wanted to mention that Fedora rawhide has 3.9 already and the test suite passes with that version. :-)

**Comment by bmw** [comment]
This should be possible once https://github.com/actions/virtual-environments/issues/1740 is resolved.

**Comment by bmw** [comment]
This should be unblocked because Python 3.9 should be available in the Linux images on Azure Pipelines.

**Comment by bmw** [comment]
I made quite a bit of progress on this since I started working on it on Wednesday. What I currently have is in the [test-py39](https://github.com/certbot/certbot/tree/test-py39) branch which builds on https://github.com/certbot/certbot/pull/8401. You can see all tests passing besides `mypy` at https://dev.azure.com/certbot/certbot/_build/results?buildId=2893&view=results.

There's more work to be done though. I had to update `dnspython` to 2.0+ since previous versions do not support Python 3.9 which created new work for us. I think we need to:

- [ ] Fix https://github.com/certbot/certbot/issues/7806. Newer versions of `dnspython` uses f-strings and [we configure `mypy` to parse files as Python 2 code](https://github.com/certbot/certbot/blob/4ea98d830bcc3d1b980a4055243c6a6a25d8dc54/mypy.ini#L4) which doesn't support f-strings causing it to fail. I'd recommend doing this in another PR.
- [ ] Since we're upgrading to a new major version of dnspython with some (what appear to be quite minor) API changes, I think we should manually run integration tests with `certbot-dns-rfc2136` and the new version of `dnspython` to make sure it still works before we pin that version in our snaps and Docker images. I have never done this before.

I think the most useful thing someone else could do is review https://github.com/certbot/certbot/pull/8401, however, if someone wants to work on these pieces, feel free to do so. I encourage people to update the issue if they do so we know who is doing what.
